File: 11-Bridge/src/java/chapter11/Device.java
package chapter11;

/**
 * Implementation interface — low-level device operations.
 * This is one side of the bridge.
 */
public interface Device {
    boolean isEnabled();
    void enable();
    void disable();
    int getVolume();
    void setVolume(int volume);
    int getChannel();
    void setChannel(int channel);
    String getName();
}

File: 11-Bridge/src/rust/device.rs
/// Implementation interface — low-level device operations.
pub trait Device {
    fn is_enabled(&self) -> bool;
    fn enable(&mut self);
    fn disable(&mut self);
    fn get_volume(&self) -> i32;
    fn set_volume(&mut self, volume: i32);
    fn get_channel(&self) -> i32;
    fn set_channel(&mut self, channel: i32);
    fn get_name(&self) -> &str;
}
